install/src/Cabal.hs

```diff
@@ -129,3 +133,15 @@ requiredCabalVersion | isWindowsSystem = requiredCabalVersionForWindows
 
 requiredCabalVersionForWindows :: RequiredVersion
 requiredCabalVersionForWindows = [3, 0, 0, 0]
+
+
+defaultProjectFile :: FilePath
+defaultProjectFile = "cabal.project"
+
+getCabalProjectFile :: VersionNumber -> Action FilePath
+getCabalProjectFile versionNumber = do
+  let versionedProjectFile = "cabal.project." <> versionNumber
```

Comment:
```suggestion
  let versionedProjectFile = "cabal.project." <> versionNumber <> ".freeze"
```
With that we could ride of dummy `cabal.project.${version}`?

Reply:
I don't understand

Reply:
I mean, if we tell cabal to use directly the freeze file in the script f.e.: `cabal install --project-file cabal.project.8.6.5.freeze` we would not need to use the freeze file indirectly with `cabal install --project-file cabal.project.8.6.5` and we could remove them, as afaiu they are alias for `cabal.project` without any meaningful content.

Reply:
> I mean, if we tell cabal to use directly the freeze file in the script

Then you ignore the actual project file and it will not build (try it yourself). Freeze files are not for general project configuration, they are for pinning dependencies.

Reply:
Mmm i guess they are auxiliary files to make `cabal v2-freeze -w ghc-8.6.5 --project-file cabal.project.8.6.5`

Reply:
`cabal v2-freeze -w ghc-8.6.5 --project-file cabal.project.8.6.5` generates `cabal.project.8.6.5`.

See the documentation: https://www.haskell.org/cabal/users-guide/nix-local-build.html#cabal-v2-freeze

Reply:
Well i would prefer to do `cabal v2-freeze -w ${ghcVersion} && mv cabal.project.freeze cabal.project.${ghcVersion}.freeze` than to have those files but i am not strongly against the original option.
What do you think @fendor?

Reply:
> Well i would prefer to do `cabal v2-freeze -w ${ghcVersion} && mv cabal.project.freeze cabal.project.${ghcVersion}.freeze` than to have those files permanently but i am not strongly against the original option.

I still don't understand what you are saying. If you want to use `cabal.project.${ghcVersion}.freeze`, then you **must** have `cabal.project.${ghcVersion}`. There is no way around it.

Reply:
> Then you ignore the actual project file and it will not build (try it yourself). Freeze files are not for general project configuration, they are for pinning dependencies.

oh i see, thanks for the info, if they are needed i resolve this